applications/vanilla/models/class.categorymodel.php

```diff
@@ -153,25 +166,93 @@ private static function loadAllCategories() {
     }
 
     /**
-     * @param $Category
+     * Calculate the user-specific information on a category.
+     *
+     * @param array &$category The category to calculate.
      */
-    private static function calculateUser(&$Category) {
-        $Category['Url'] = url($Category['Url'], '//');
-        if ($Photo = val('Photo', $Category)) {
-            $Category['PhotoUrl'] = Gdn_Upload::url($Photo);
+    private function calculateUser(&$category, $addUserCategory = null) {
+        $category['Url'] = url($category['Url'], '//');
```

Comment:
In its current state, this line triggers a notice when viewing a discussion:

> Undefined index: Url

`Url` isn't available in `$category` at this point, but `UrlCode` is.  This notice isn't seen on current master.  It looks like this particular element may've been previously inserted with `CategoryModel::calculate`.